entity/did_test.go

```diff
@@ -0,0 +1,37 @@
+package entity
+
+import (
+	"testing"
+
+	"github.com/mr-tron/base58"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/crypto/sha3"
+)
+
+func TestNewDID(t *testing.T) {
+	// TODO should NewDID() validate the kind of key passed?
+	testBytes := []byte("foo")
+
+	// make a SHA from the testBytes, grab the last 20 and prefix \x16
+	testHash := sha3.Sum256(testBytes)
+	testTruncHash := testHash[len(testHash)-20:]
```

Comment:
this is an interesting test since test since testBytes above is smaller the 20 bytes. lets also have a case for the happy path.

Reply:
But the SHA that we truncate is 32 :)

Reply:
Ah, maybe you meant this comment for the non-happy path test below? `TestNewDID` is meant to test good path.